pybettor/implied_odds.py

```diff
@@ -1,20 +1,192 @@
+from typing import Union
 from fractions import Fraction
+from scipy import optimize
+import numpy as np
 
 
-def implied_odds(prob, category: str = "us") -> list or dict:
+def _implied_naive_odds(prob, category):
+    if category == "all":
+        us = [x / (1 - x) * -100 if x > 0.5 else (1 - x) / x * 100 for x in prob]
+        dec = [round(1 / x, 2) for x in prob]
+        frac = [Fraction(x - 1).limit_denominator(100) for x in dec]
+        frac = [str(x.numerator) + "/" + str(x.denominator) for x in frac]
+        prob = prob
+        imp_odds = {
+            "American": us,
+            "Decimal": dec,
+            "Fraction": frac,
+            "Implied Probability": prob,
+        }
+
+    elif category == "us":
+        imp_odds = [x / (1 - x) * -100 if x > 0.5 else (1 - x) / x * 100 for x in prob]
+        imp_odds = [round(x) for x in imp_odds]
+
+    elif category == "dec":
+        imp_odds = [round(1 / x, 2) for x in prob]
+
+    elif category == "frac":
+        imp_odds = [Fraction((1.0 / x) - 1.0).limit_denominator(100) for x in prob]
+        imp_odds = [str(x.numerator) + "/" + str(x.denominator) for x in imp_odds]
+
+    return imp_odds
+
+
+def _implied_basic_odds(prob, margin):
+    return [1 / (x * (1 + margin)) for x in prob]
```

Comment:
this does not return odds in the right format. I didn't go through each method but i'm pretty positive that each method returns decimal odds. We'd need to add an addition line to convert decimal odds to whatever category is in the input for the function

Reply:
this should be fixed now